services/llm_service.py

```
@@ -0,0 +1,261 @@
+                model=model_name,
+                google_api_key=SecretStr(self.api_key),
+                temperature=generation_config["temperature"],
+                top_p=generation_config["top_p"],
+                top_k=generation_config["top_k"],
+                max_output_tokens=generation_config["max_output_tokens"],
+                safety_settings=safety_settings,
+                convert_system_message_to_human=True
+            )
+            
+        except Exception as e:
+            raise ExternalAPIError(f"Gemini LLMの作成に失敗しました: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)